foundations/javascript_basics/understanding_errors.md

```diff
@@ -0,0 +1,132 @@
+# Introduction
+
+Reading error messages is an underrated art form. At first glance, many beginners shrink away from error messages as they appear to be “scary” and difficult to understand because they include terms one may not be familiar with. However, error messages provide developers with a treasure trove of knowledge, and tell you everything you need to know about how to resolve them! Being able to parse error messages and warnings without fear will enable you to be able to effectively debug your applications, receive meaningful help from others, and empower yourself to push forward when faced with an error.
+
+# Learning Outcomes
+
+- Name at least three kinds of Javascript errors
+- Identify two parts of an error message that help you find where the error originates
+- Be able to understand how to research and resolve errors
+
+# The Anatomy of an Error
+
+Errors contain crucial information that can assist you in locating the code responsible for the error, determining why you have this error, and resolving the error. Let’s assume we have written the following code:
```

Comment:
We should mention that people can run this code, and should, in their console in the browser.